Rebus/Workers/ThreadBased/ThreadWorker.cs

```diff
@@ -183,14 +187,18 @@ void DoWork(bool onlyRunContinuations = false)
             }
         }
 
-        async Task<ReceiveResult> TryReceiveTransportMessage(DefaultTransactionContext transactionContext)
+        async Task<ReceiveResult> TryReceiveTransportMessage(DefaultTransactionContext transactionContext, CancellationToken cToken)
         {
             try
             {
-                var message = await _transport.Receive(transactionContext);
+                var message = await _transport.Receive(transactionContext, cToken);
 
                 return new ReceiveResult(message);
             }
+            //catch (TaskCanceledException tex)
+            //{
+            //    return new ReceiveResult(tex, true);
+            //}
```

Comment:
Yep. I wanted to make a special treatment for cancelation but for now, a simple log warning works :)